### PR TITLE
[CWS] simplify a bit the mount resolver

### PR DESCRIPTION
**Pull request by safchain**
### What does this PR do?

Do not use anymore deleted mount entry in case of reorderer usage.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/security/resolvers/mount/resolver.go [hunk 1]
```diff
@@ -33,8 +32,14 @@ const (
 	deleteDelayTime                      = 5 * time.Second
 	numAllowedMountIDsToResolvePerPeriod = 75
 	fallbackLimiterPeriod                = 5 * time.Second
+	redemptionTime                       = 5 * time.Second
 )
 
+type redemptionEntry struct {
```
Comment (modernplumbing) [inline pkg/security/resolvers/mount/resolver.go:38]: what does redemption mean in this context?

File: pkg/security/resolvers/mount/resolver.go [hunk 2]
```diff
@@ -232,10 +236,10 @@ func (mr *Resolver) Insert(e model.Mount) error {
 func (mr *Resolver) insert(m *model.Mount) {
 	// umount the previous one if exists
 	if prev, ok := mr.mounts[m.MountID]; ok {
-		// if present in the redemption that the evict function that will remove the entry
-		if present := mr.redemption.Remove(prev.MountID); !present {
-			mr.finalize(prev)
-		}
+		mr.finalize(prev)
+	} else if prev, ok := mr.redemption.Get(m.MountID); ok {
```
Comment (lebauce) [inline pkg/security/resolvers/mount/resolver.go:240]: also check opts.UseRedemption

File: pkg/security/resolvers/mount/resolver.go (hunk 1 above)
Comment (safchain) [reply pkg/security/resolvers/mount/resolver.go:38]: Add a chance for a misordered event (within 5sec) to get a correct mount resolution

File: pkg/security/resolvers/mount/resolver.go [hunk 3]
```diff
@@ -257,14 +267,25 @@ func (mr *Resolver) insert(m *model.Mount) {
...
-	mount, exists := mr.mounts[mountID]
-	if !exists {
-		return "", &ErrMountNotFound{MountID: mountID}
+	mount := mr.mounts[mountID]
+	if mount == nil && mr.opts.UseRedemption {
+		mount = mr.getFromRedemption(mountID)
+		if mount == nil {
+			return "", &ErrMountNotFound{MountID: mountID}
+		}
 	}
```
Comment (YoannGh) [inline pkg/security/resolvers/mount/resolver.go:290]:
```suggestion
	mount := mr.mounts[mountID]
	if mount == nil {
	    if mr.opts.UseRedemption {
		    mount = mr.getFromRedemption(mountID)
		    if mount == nil {
			    return "", &ErrMountNotFound{MountID: mountID}
		    }
		} else {
		    return "", &ErrMountNotFound{MountID: mountID}
		}
	}
```

**Review by YoannGh — APPROVED** [review]

**Review by paulcacheux — APPROVED** [review]

**Pull request merged**